### ISSUE TITLE
Key inside a json value is replaced by columnName of attribute

### ISSUE THREAD

**Issue by Bangsheng** [balderdashy/waterline#1313]
I have a model defined as:
```
{
  ...
 phone: {
    type: 'string',
    columnName: 'ph'
  },
  prefer_ship_address: {
    type: 'json',
    columnName: 'ps'
  },
...
}
```

When I try to update prefer_ship_address, I find the key name 'phone' inside the value of the prefer_ship_address is replaced to 'ph'. For example, if I update prefer_ship_address with value
```
{
  name: 'foo',
  address: 'bar',
  phone: '11111111111'
}
```
, then I print input `values` of `update` function in the adapter
```
{
  updatedAt: Mon Feb 22 2016 14:25:32 GMT-0600 (CST),
  ps: 
    {
      name: 'foo',
      address: 'bar',
      ph: '1111111111'
    }
}
```

**Comment by atiertant** [comment]
that's true, adapter interact with database,so it only need to know columnName and waterline transform results in model objects.

**Comment by atiertant** [comment]
looks like an adapter problem,which one do you use?

**Comment by Bangsheng** [comment]
@atiertant I use my own adapter. But I don't think this is an adapter problem, because I print out `values` at the first line in the `update` function. That means adapter has not done anything yet to `values` which is passed by waterline.
```
update: function(connectionName, collectionName, options, values, cb) {
      console.log(values);
...
```

**Comment by particlebanana** [comment]
Looks like it's recursively replacing instances of `phone` with the column name. Could you post the update query you are using or create a repo with the models and a route that re-creates the issue?

**Comment by Bangsheng** [comment]
@particlebanana Here is the `update` query.
```
sails.models.user.update(
{
  where: {
    uid: 'xxxxxxx'
  }
}, 
{
  prefer_ship_address: 
  {
    name: 'foo',
    address: 'bar',
    phone: '11111111111' 
  }
});

**Comment by atiertant** [comment]
@Bangsheng i think [offshore PR-17](https://github.com/Atlantis-Software/offshore/pull/17) should correct this. could you test with current offshore master branch ?

**Comment by Bangsheng** [comment]
@atiertant I tested with offshore@0.0.3 and the problem was not solved.

**Comment by atiertant** [comment]
@Bangsheng the fix is not in npm@0.0.3 only on git for now. could you replace your offshore/lib/offshore/core/transformations.js with this one [transformations.js](https://raw.githubusercontent.com/Atlantis-Software/offshore/master/lib/offshore/core/transformations.js)?

**Comment by Bangsheng** [comment]
@atiertant Yes! Problem Solved. Looks like `function recursiveParse` inside `Transformation.prototype.serialize`  needs some changes. Brilliant! Thanks!

**Comment by wulfsolter** [comment]
Wow, this is breaking out data in production!

@particlebanana : Simple repo with reproduction https://github.com/wulfsolter/waterline-issue-json-attributes-changing

Current workaround is use the Offshore transformations.js linked above.
